ksqldb-execution/src/main/java/io/confluent/ksql/execution/windows/TumblingWindowExpression.java

```diff
@@ -21,44 +21,37 @@
 import io.confluent.ksql.model.WindowType;
 import io.confluent.ksql.parser.NodeLocation;
 import io.confluent.ksql.serde.WindowInfo;
-import java.time.Duration;
 import java.util.Objects;
 import java.util.Optional;
-import java.util.concurrent.TimeUnit;
 
 @Immutable
 public class TumblingWindowExpression extends KsqlWindowExpression {
 
-  private final long size;
-  private final TimeUnit sizeUnit;
+  private final WindowTimeClause size;
 
-  public TumblingWindowExpression(final long size, final TimeUnit sizeUnit) {
-    this(Optional.empty(), size, sizeUnit);
+  public TumblingWindowExpression(final WindowTimeClause size) {
+    this(Optional.empty(), size, Optional.empty(), Optional.empty());
   }
 
   public TumblingWindowExpression(
       final Optional<NodeLocation> location,
-      final long size,
-      final TimeUnit sizeUnit
+      final WindowTimeClause size,
+      final Optional<WindowTimeClause> retention,
+      final Optional<WindowTimeClause> gracePeriod
   ) {
-    super(location);
-    this.size = size;
-    this.sizeUnit = requireNonNull(sizeUnit, "sizeUnit");
+    super(location, retention, gracePeriod);
+    this.size = requireNonNull(size, "size");
   }
 
   @Override
   public WindowInfo getWindowInfo() {
     return WindowInfo.of(
         WindowType.TUMBLING,
-        Optional.of(Duration.ofNanos(sizeUnit.toNanos(size)))
```

Comment:
this is what ends up being included in the query plan, do we need to encode the `retention` and `gracePeriod` here?

Reply:
I would like to context on this. I may not be fully understanding this. If this will affect recoverability i.e take the cluster down and when queries are recreated, then we probably need to..

Reply:
@rodesai as well.. 
Since rohan confirmed that the window expression is what we use, added cases in SqlFormatterTest around the two clauses.. They are working as expected.. (parsing, formatting)